resolwe_bio/tools/samplehcluster.py

```diff
@@ -53,36 +49,35 @@ def get_expressions(fnames, sep='\t', gene_set=[]):
 
     """
     dfs = [get_expression(fname, sep=sep, gene_set=gene_set) for fname in fnames]
-    df = pd.concat(dfs, axis=1)
-    return df.dropna()
+    inner = pd.concat(dfs, axis=1, join='inner')
+    outer = pd.concat(dfs, axis=1, join='outer', sort=True)
+    if gene_set:
+        excluded = sorted(set(gene_set).difference(set(inner.index)))
+    else:
+        excluded = sorted(outer.index.difference(inner.index))
+    return inner, excluded
 
 
-def transform(expressions, log2=False, const=1.0, normalization=None, ddof=1):
+def transform(expressions, log2=False, const=1.0, z_score=False, ddof=1):
     """Compute log2 and normalize expression values.
 
     Parameters:
+    - log2: use log2(x+const) transformation
     - const: an additive constant used in computation of log2
-    - normalization: None or 'z-score'
-    - ddof: degrees of freedom used in computation of z-scores
+    - z_score: use Z-score normalization
+    - ddof: degrees of freedom used in computation of Z-score
 
     """
     if log2:
-        try:
-            expressions = expressions.applymap(lambda x: np.log2(x + const))
-        except:
+        expressions = expressions.applymap(lambda x: np.log2(x + const))
+        if expressions.isnull().values.any():
             msg = 'Cannot apply log2 to expression values.'
             print(error(msg))
             raise ValueError(msg)
```

Comment:
I don't see any value in raising ValueError after `print(error(msg))`. Instead of repeating the same message with `raise ValueError(msg)`, just use `sys.exit(1)`

Reply:
Same for many examples below.

Reply:
Since there are so many lines of form:

    print(error(msg))
    raise ValueError(msg) / sys.exit(1)

consider defining function to do this operation in one line...